src/core/plot/qgsplot.h

```diff
@@ -0,0 +1,438 @@
+    std::unique_ptr< QgsLineSymbol > mGridMinorSymbol;
+
+    QgsTextFormat mLabelTextFormat;
+
+};
+
+/**
+ * \brief Base class for 2-dimensional plot/chart/graphs.
+ *
+ * The base class is responsible for rendering the axis, grid lines and chart area. Subclasses
+ * can implement the renderContent() method to render their actual plot content.
+ *
+ * \warning This class is not considered stable API, and may change in future!
+ *
+ * \ingroup core
+ * \since QGIS 3.26
+ */
+class CORE_EXPORT Qgs2DPlot : public QgsPlot
+{
+  public:
+
+    /**
+     * Constructor for Qgs2DPlot.
+     */
+    Qgs2DPlot();
+
+    ~Qgs2DPlot() override;
+
+    //! Qgs2DPlot cannot be copied
+    Qgs2DPlot( const Qgs2DPlot &other ) = delete;
+    //! Qgs2DPlot cannot be copied
+    Qgs2DPlot &operator=( const Qgs2DPlot &other ) = delete;
+
+    bool writeXml( QDomElement &element, QDomDocument &document, QgsReadWriteContext &context ) override;
+    bool readXml( const QDomElement &element, QgsReadWriteContext &context ) override;
+
+    /**
+     * Renders the plot.
+     */
+    void render( QgsRenderContext &context );
+
+    /**
+     * Renders the plot content.
+     *
+     * Subclasses can implement this method to render the actual plot content (e.g. bar charts, scatter plots).
+     * This method will be called after the chart background and grid are rendered, but before the chart border is rendered.
+     *
+     * The default implementation does nothing.
+     *
+     * The \a plotArea argument specifies that area of the plot which corresponds to the actual plot content. Implementations
+     * should take care to scale values accordingly to render points correctly inside this plot area.
+     */
+    virtual void renderContent( QgsRenderContext &context, const QRectF &plotArea );
+
+    /**
+     * Returns the overall size of the plot (in millimeters) (including titles and other components which sit outside the plot area).
+     *
+     * \see setSize()
+     */
+    QSizeF size() const;
+
+    /**
+     * Sets the overall \a size of the plot (including titles and over components which sit outside the plot area).
+     *
+     * \see size()
+     */
+    void setSize( QSizeF size );
+
+    /**
+     * Returns the area of the plot which corresponds to the actual plot content (excluding all titles and other components which sit
+     * outside the plot area).
+     */
+    QRectF interiorPlotArea( QgsRenderContext &context ) const;
+
+    /**
+     * Automatically sets the grid and label intervals to optimal values
+     * for display in the given render \a context.
+     *
+     * Intervals will be calculated in order to avoid overlapping axis labels and to ensure
+     * round values are shown.
+     */
+    void calculateOptimisedIntervals( QgsRenderContext &context );
+
+    /**
+     * Returns the minimum value of the x axis.
+     *
+     * \see setXMinimum()
+     */
+    double xMinimum() const { return mMinX; }
```

Comment:
shouldn't the min/max be part of the axis?

Reply:
I debated that, but at this stage I'd prefer to keep the axis class more generic. I'm not sure that we'll always have a range associated with an axis, eg for a polar plot or some other form. I've marked the API as non stable so we have the ability to refine later as we gain a better idea of the various use cases.

Reply:
I still think it's more future-proof and coherent. To me, the most valid argument would be the multi-axis plot, otherwise you'd have to have multiple plot areas?